contrib/natss/pkg/dispatcher/dispatcher/dispatcher_test.go

```diff
@@ -106,6 +115,54 @@ func TestSubscribeUnsubscribe(t *testing.T) {
 	}
 }
 
+func TestMalformedMessage(t *testing.T) {
+	logger.Info("TestMalformedMessage()")
+
+	cRef := provisioners.ChannelReference{Namespace: "test_namespace", Name: "test_channel"}
+	sRef := subscriptionReference{Name: "sub_name", Namespace: "sub_namespace", SubscriberURI: "", ReplyURI: ""}
+
+	// subscribe to a channel
+	if _, err := s.subscribe(cRef, sRef); err != nil {
+		t.Errorf("Subscribe to NATSS failed: %v", err)
+	}
+	defer func() {
+		if err := s.unsubscribe(cRef, sRef); err != nil {
+			t.Errorf("Close subscription to NATSS failed: %v", err)
+		}
+	}()
+
+	m := &provisioners.Message{
+		Headers: map[string]string{"header1": "value1", "header2": "value2"},
+		Payload: []byte{'1', '2', '3', '4', '5'},
+	}
+	ch := getSubject(cRef)
+	message, err := json.Marshal(m)
+	if err != nil {
+		t.Errorf("Error during marshaling of the message: %v", err)
+	}
+	// Corrupting message so it would fail Unmarshal
+	b := message[:len(message)-3]
+	if err := stanutil.Publish(s.natssConn, ch, &b, testLogger.Sugar()); err != nil {
+		logger.Errorf("Error during publish: %v", err)
+		t.Errorf("Error during publish: %v", err)
+	}
+	// Need to wait until the messages reaches mcb() where Unmarshal takes place
+	timeout := time.After(20 * time.Second)
+	ticker := time.NewTicker(1 * time.Second)
```

Comment:
This is perfectly fine; another option if nats is ordered would be to send an uncorrupted message after the corrupted one, and check for the second message being delivered. But this works fine, thanks for writing the test!